tests/stdlib/tasynchttpserver_transferencoding.nim

```diff
@@ -1,42 +1,56 @@
+discard """
+  matrix: "--gc:arc --threads:on; --gc:arc --threads:on -d:danger; --threads:on"
+"""
+
 import httpclient, asynchttpserver, asyncdispatch, asyncfutures
 import net
 
 import std/asyncnet
 import std/nativesockets
+import std/threadpool
 
 const postBegin = """
 POST / HTTP/1.1
 Transfer-Encoding:chunked
 
 """
 
-template genTest(input, expected) =
-  var sanity = false
-  proc handler(request: Request) {.async.} =
+template genTest(input, expected: string) =
+  proc handler(request: Request, future: Future[bool]): Future[void] {.async, gcsafe.} =
       doAssert(request.body == expected)
       doAssert(request.headers.hasKey("Transfer-Encoding"))
       doAssert(not request.headers.hasKey("Content-Length"))
-      sanity = true
+      future.complete(true)
       await request.respond(Http200, "Good")
 
-  proc runSleepLoop(server: AsyncHttpServer) {.async.} = 
+  proc runSleepLoop(server: AsyncHttpServer, future: Future[bool]) {.async, gcsafe.} = 
+    proc wrapper(request: Request): Future[void] {.gcsafe, closure.} =
```

Comment:
```suggestion
    proc wrapper(request: Request) {.gcsafe, closure.} =
```

Reply:
interesting, here it does't here, but it works in https://github.com/nim-lang/Nim/pull/18052/files#r641657077

Reply:
Fixed.